assets/scripts/menus/LocaleDropdown.jsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { t, getLocale, onNewLocaleSelected } from '../app/locale'
+import { trackEvent } from '../app/event_tracking'
+
+/**
+ * Locale levels
+ * Locales are grouped into four levels of "readiness" for production.
+ *
+ * 1 - Incomplete or in progress, available for translators to test, but not for end-users.
+ * 2 - Complete or nearly complete, available for quality assurance & feedback with small group of users.
+ * 3 - Complete and ready for production use.
+ * 4 - This is a special category for the "default" locale, which is English. This should never
+ *     be disabled and we don't have a special flag to toggle group 4 on or off. It's always on.
+ *
+ * Levels are lower bounds and inclusive. In other words, allowing level 1 means all locales
+ * 1 or higher are enabled. You cannot _only_ turn on level 1 but not level 2 for example.
+ */
+const LOCALES = [
+  {
+    label: 'Chinese (Traditional)',
+    value: 'zh-Hant',
+    key: 'i18n.lang.zh-hant',
+    level: 1
+  },
+  {
+    label: 'English',
+    value: 'en',
+    key: 'i18n.lang.en',
+    level: 4
+  },
+  {
+    label: 'Finnish',
+    value: 'fi',
+    key: 'i18n.lang.fi',
+    level: 2
+  },
+  {
+    label: 'French',
+    value: 'fr',
+    key: 'i18n.lang.fr',
+    level: 2
+  },
+  {
+    label: 'German',
+    value: 'de',
+    key: 'i18n.lang.de',
+    level: 1
+  },
+  {
+    label: 'Polish',
+    value: 'pl',
+    key: 'i18n.lang.pl',
+    level: 1
+  },
+  {
+    label: 'Portuguese (Brazil)',
+    value: 'pt_BR',
+    key: 'i18n.lang.pt-br',
+    level: 1
+  },
+  {
+    label: 'Spanish',
+    value: 'es',
+    key: 'i18n.lang.es',
+    level: 1
+  },
+  {
+    label: 'Spanish (Mexico)',
+    value: 'es_MX',
+    key: 'i18n.lang.es-mx',
+    level: 2
+  },
+  {
+    label: 'Swedish',
+    value: 'sv',
+    key: 'i18n.lang.sv',
+    level: 1
+  }
+]
+
+export class LocaleDropdown extends React.Component {
+  static propTypes = {
+    /* eslint-disable react/no-unused-prop-types */
+    // These props _are_ used but linter can't tell
+    level1: PropTypes.bool.isRequired,
+    level2: PropTypes.bool.isRequired,
+    level3: PropTypes.bool.isRequired
+    /* eslint-enable react/no-unused-prop-types */
+  }
+
+  static defaultProps = {
```

Comment:
Do we need defaultProps if we all propTypes are set as required?

Reply:
It's a little overkill, but it's there to guarantee behavior.